library/src/error/flatten/flatten.ts

```diff
@@ -78,7 +78,7 @@ type NestedPath<TSchema extends BaseSchema | BaseSchemaAsync> =
     TSchema extends
         | ObjectSchema<infer TEntries, infer TRest>
         | ObjectSchemaAsync<infer TEntries, infer TRest>
-    ? TRest extends BaseSchema | BaseSchemaAsync
```

Comment:
Because this change may be ambiguous, the type condition we have here is attempting to remove `undefined` from the union of types that `TRest` is assignable to. because `TRest` is inferred from the previous conditional. The utility type `NonNullable` takes whatever the values that `TRest` can be and removes `null` and `undefined` from that union.

It's been over a month since I had run into a type error here, so I don't remember the exact error message was, but the above ended up being the solution.